docs/specs/user-profile/design.md

```diff
@@ -0,0 +1,329 @@
+# User profile
+
+User profile consists of standard attributes and custom attributes.
+
+## Standard attributes
+
+Standard attributes is a subset of [OIDC standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims).
+All standard attributes are optional.
+All standard attributes are readable and writable via the Admin API.
+All standard attributes are readable and writable in the portal.
+
+All standard attributes are by default readable and writable by the end-user in the settings page.
+The developer can restrict access to individual standard attribute.
+
+```yaml
+ui:
+  standard_attributes:
+  - pointer: /given_name
+    access_control: hidden
+  - pointer: /family_name
+    access_control: hidden
+```
+
+### email
+
+The primary email address of the end-user.
+The value always comes from one of the identity the end-user has.
+Therefore, the input control is a dropdown.
+It is subject to [automatic population](#automatic-population).
+
+### email\_verified
+
+Whether the primary email address is verified.
+This attribute is present when `email` is present.
+It is NOT writable.
+
+### phone\_number
+
+The primary phone number of the end-user.
+The value always from from one of the identity the end-user has.
+Therefore, the input control is a dropdown.
+It is subject to [automatic population](#automatic-population).
+
+### phone\_number\_verified
+
+Whether the primary phone number is verified.
+This attribute is present when `phone_number` is present.
+It is NOT writable.
+
+### preferred\_username
+
+The primary username of the end-user.
+The value always comes from one of the identity the end-user has.
+Therefore, the input control is a dropdown.
+It is subject to [automatic population](#automatic-population).
+
+### zoneinfo
+
+The preferred tz database name, e.g. `Asia/Hong_Kong`.
+The list of tz database names are constants, so the input control is a dropdown.
+
+### locale
+
+The preferred BCP47 tag, e.g. `zh-HK`.
+The input control is a dropdown, and the options are the supported languages of the project.
+
+### birthdate
+
+The supported format is `YYYY-MM-DD`, representing the birthdate, e.g. 1 Jan 1992.
+
+> The OIDC spec also allows `0000-MM-DD` for birthday and `YYYY` for year of birth.
+> We only support `YYYY-MM-DD` for simplicity.
+
+### given\_name
+
+The given name of the first name of the end-user.
+The input control is a simple text input without validation.
+It is subject to [automatic population](#automatic-population).
+
+### family\_name
+
+The family name or the last name of the end-user.
+The input control is a simple text input without validation.
+It is subject to [automatic population](#automatic-population).
+
+## Automatic population
+
+> Automatic population CANNOT be turned off!
```

Comment:
I guess you might misunderstand my proposal on the automatic population.
The standard attributes subject to the automatic population are only populated when they are originally absent.
So if the end-user already has the email attribute present, it will not be changed because a new identity is added.